portindex2json/portindex2json.tcl

```diff
@@ -12,6 +12,74 @@
 
 package require json::write
 
+proc format_strings {strings} {
+    set new_strings {}
+    foreach element $strings {
+        lappend new_strings [::json::write string $element]
+    }
+    return $new_strings
+}
+
+proc get_maintainer_array {maintainer} {
+    set detail_list [split $maintainer @]
+    if {[string first : [lindex $detail_list 0]] != -1} {
+        set email_list [split [lindex $detail_list 0] :]
+        set name [string trimright [lindex $email_list 1]]
+        set email(name) "\"$name\""
+        set email(domain) "\"[lindex $email_list 0]\""
+        ::json::write indented false
+        set email_subobject [::json::write object {*}[array get email]]
+        ::json::write indented true
+    } else {
+        set name [string trimright [lindex $detail_list 0]]
+        set email(name) "\"$name\""
+        set email(domain) "\"macports.org\""
+        ::json::write indented false
+        set email_subobject [::json::write object {*}[array get email]]
+        ::json::write indented true
+    }
+
+    set maintainer_array(github) "\"[lindex $detail_list 1]\""
```

Comment:
I would be kind of inclined to either leave this out completely or maybe set it to `null` if the github handle doesn't exist. That said, we should probably get a warning whenever a github handle is missing (probably in application rather than here), so in a way making it explicit that it doesn't exist (either with an empty string or with `null`) is still informative if one wants to search through the json file itself. Second opinion welcome.

And please use json function for quoting.

Reply:
I have set it use null if the GitHub handle is not provided.